### PR TITLE
Fix functional tests

### PR DESCRIPTION
**Pull request by devimc**
This patch duplicates error and critical messages
to strerr and log file, before this patch running
cor standalone with wrong options and parameters
does not report any error to user

Signed-off-by: Julio Montes <julio.montes@intel.com>

### CODE REVIEW

**Comment by dlespiau** [comment]
Makes sense to me, but I don't know the history there.

lgtm

**Comment by jcvenegas** [comment]
lgtm

**Comment by chavafg** [comment]
qa-failed

**Comment by devimc** [comment]
I'll fix all functional tests

**Comment by sameo** [comment]
LGTM

**Comment by sameo** [comment]
@devimc Why is this still a `do-not-merge` ?

**Comment by devimc** [comment]
Hi @sameo , some functional tests are still failing and cc-shim never dies #455

**Comment by devimc** [comment]
I think that before merge this PR, #475 should be fixed

**Comment by sboeuf** [comment]
@devimc , are you good to get this PR merged ? And @chavafg could we have a new QA testing on this one ?

**Comment by devimc** [comment]
@sboeuf not yet, to run these functional tests is needed to have cc-shim installed , I wouldn't like to have this dependency

File: tests/functional/common.bash.in [hunk 1]
```diff
@@ -220,9 +205,9 @@ function testcontainer() {
 
 	while [ "$i" -lt "$timeout_cmd" ]
 	do
-		local output=$(run_cmd "$cmd")
-		[[ ${output} =~ ID\ +PID\ +STATUS\ +BUNDLE\ +CREATED+ ]]
-		if [[ "${output}" == *"$container_id"*[0-9]*"${check_status}"*"$BUNDLE_DIR"* ]]
```
Comment (jcvenegas) [inline tests/functional/common.bash.in]: How do you check the container state now ?

File: proxy/proxy.go [hunk 2]
```diff
@@ -252,11 +252,19 @@ func newProxy() *proxy {
 
 // This variable is populated at link time with the value of:
 //   ${locatestatedir}/run/cc-oci-runtime/proxy
-var socketPath string
+var DefaultSocketPath string
```
Comment (dlespiau) [inline proxy/proxy.go]:
```
cc-oci-runtime-0.1/proxy/proxy.go:253:1: comment on exported var DefaultSocketPath should be of the form "DefaultSocketPath ..."
cc-oci-runtime-0.1/proxy/proxy.go:257:1: comment on exported var ArgSocketPath should be of the form "ArgSocketPath ..."
```

You could either change the comments to start like go comments golint expects, or (better IMO), don't expose those variables outside of the package.

Comment (sameo) [inline proxy/proxy.go]: You need to fix the comment here. In golang, an exported variable need to be described with a comment starting with the variable name itself. So in that case:

```
// DefaultSocketPath is populated....
```

File: proxy/proxy.go [hunk 3]
```diff
@@ -252,11 +252,19 @@ func newProxy() *proxy {
...
+
+// This variable is populated at runtime from the option -socket-path
+var ArgSocketPath = flag.String("socket-path", "", "specify path to socket file")
```
Comment (sameo) [inline proxy/proxy.go]: Ditto

**Comment by dlespiau** [comment]
I wish the big commit with lots of changes had been split in smaller pieces, it's pretty much un-reviewable in this form. That something easy to fix for your following PRs! Anyway, this PR is probably better merged than not.

lgtm

File: tests/functional/common.bash.in [hunk 4]
```diff
@@ -220,9 +205,9 @@ function testcontainer() {
...
+		local output=$(run_cmd "$cmd | grep ${container_id} | grep ${check_status}")
+		echo $output >> /tmp/LOG1
```
Comment (devimc) [inline tests/functional/common.bash.in]: ups.. fixing..

File: tests/functional/common.bash.in (hunk 1 above)
Comment (devimc) [inline tests/functional/common.bash.in]: local output=$(run_cmd "$cmd | grep ${container_id} | grep ${check_status}")
if [ -n "${output}" ]

Comment (jcvenegas) [inline tests/functional/common.bash.in]: oh I see, thanks :)

**Comment by jcvenegas** [comment]
lgtm  , just move  /tmp/LOG1 to tests dir

File: tests/functional/common.bash.in [hunk 5]
```diff
@@ -386,6 +386,11 @@ function setup_common(){
 
 # Cleanup anything generated by this script
 function cleanup_common(){
+	# kill shim if it is still running
```
Comment (dlespiau) [inline tests/functional/common.bash.in]: That will hide bugs, don't think we should do that. We could warn instead?

Comment (devimc) [inline tests/functional/common.bash.in]: I don't think so, each tests calls to ```kill``` and ```delete``` if something fails will be reported
after that no cc-shim process should be running

Comment (sameo) [inline tests/functional/common.bash.in]: @devimc Agreed, we should at least make that a warning. If cc-shim is still running at that point, this is a bug we should know about.

**Comment by dlespiau** [comment]
lgtm

**Comment by sboeuf** [comment]
Sounds good to me.
LGTM

**Comment by chavafg** [comment]
qa-failed

**Comment by sboeuf** [comment]
@chavafg this PR just got merged. What was the problem about QA ?

**Comment by chavafg** [comment]
not ok 1 Run shell echo

**Pull request merged**